lib.py

```diff
@@ -0,0 +1,305 @@
+import time
+from telebot import types
+import yaml
+from pathlib import Path
+import logging
+from itertools import zip_longest
+import random
+logger = logging.getLogger(__name__)
+import threading
+lock = threading.Lock()
+
```

Comment:
We need another mutex for preventing a race condition, where one user could prevent another from registering/deleting herself. (comment 1/5)
```suggestion
file_lock = threading.Lock()
user_independence_lock = threading.Lock()
```